python/dolfinx/fem/petsc.py

```diff
@@ -1208,6 +1207,16 @@ def __init__(
         :math:`u=(u_0,...,u_N), v=(v_0,...,v_N)` using PETSc SNES as the
         non-linear solver.
 
+        By default, the underlying SNES solver uses PETSc's default
+        options. To use the robust combination of LU via MUMPS with
+        a basic linesearch, pass:
+
+            petsc_options = {"ksp_type": "preonly",
+                             "pc_type": "lu",
+                             "pc_factor_mat_solver_type": "mumps",
+                             "snes_linesearch_type": "basic",
```

Comment:
"basic" aka. "none", is the full Newton step, so is not the most robust choice here. PETSc's default "bt" is surely more robust.